src/program.ts

```diff
@@ -160,7 +178,23 @@ export async function evaluateJsonProgram(program: Program, onCall: (func: strin
         if (obj.hasOwnProperty("@ref")) {
             const index = obj["@ref"];
             if (typeof index === "number" && index < results.length) {
-                return results[index];
+                const path = obj["path"];
+                let value = results[index];
+                if (Array.isArray(path)) {
+                    for (const prop of path) {
+                        if (typeof prop === "string" && typeof value === "object" && value !== null && prop in value) {
+                            value = value[prop as keyof typeof value];
+                        }
+                        else if (typeof prop === "number" && Array.isArray(value) && prop < value.length) {
+                            value = value[prop];
+                        }
+                        else {
+                            return undefined;

```

Comment:
this needs some consideration. in the prompt template, there is a ref to a function that cannot return `undefined`, which I didn't fully understand the "why". But now with the path, it seems that undefined values are possible.